src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -320,74 +179,67 @@ bool AdaptDispatch::_CursorMovePosition(const std::optional<size_t> row, const s
 
     if (success)
     {
-        // handle optional parameters. If not specified, keep same cursor position from what we just loaded.
-        size_t rowActual = 0;
-        size_t columnActual = 0;
-
-        if (row)
+        // Calculate the viewport boundaries as inclusive values.
+        // srWindow is exclusive so we need to subtract 1 from the bottom.
+        const int viewportTop = csbiex.srWindow.Top;
```

Comment:
Is there any way to make use of the types `Viewport` class here? I generally have been trying to use that class and/or improve it as necessary for inclusive/exclusive problems like this and basic offsetting of rectangles.

Reply:
We could possibly use it for just the top/bottom range, but we can't use the clamping functionality, and it makes other parts of the code messier, so I'd rather leave it for a later refactor when we can make the most of the `Viewport` capabilities. See my comments below for more details.

Reply:
OK, makes sense based on the below. Resolving this for now.